server/http/src/service.rs

```diff
@@ -193,28 +142,44 @@ where
     fn handle_graphql_query(&self, id: SubgraphId, request_body: Body) -> GraphQLServiceResponse {
         let service = self.clone();
 
-        let schemas_rwlock = service.schemas.clone();
-        let schemas = schemas_rwlock.read().unwrap();
-
-        let schema_opt = schemas.get(&id).map(|s| s.to_owned());
-
-        match schema_opt {
-            None => service.handle_not_found(),
-            Some(schema) => Box::new(
-                request_body
-                    .concat2()
-                    .map_err(|_| GraphQLServerError::from("Failed to read request body"))
-                    .and_then(move |body| GraphQLRequest::new(body, schema.clone()))
-                    .and_then(move |query| {
-                        // Run the query using the query runner
-                        service
-                            .graphql_runner
-                            .run_query(query)
-                            .map_err(|e| GraphQLServerError::from(e))
-                    })
-                    .then(|result| GraphQLResponse::new(result)),
-            ),
+        match self.store.is_deployed(&id) {
+            Err(e) => {
+                return Box::new(future::err(GraphQLServerError::InternalError(
+                    e.to_string(),
+                )))
+            }
+            Ok(false) => {
+                return Box::new(future::err(GraphQLServerError::ClientError(format!(
+                    "Subgraph {} not deployed",
+                    id
+                ))))
+            }
+            Ok(true) => (),
         }
```

Comment:
Is it possible here to still allow querying of subgraphs without a deployment? I'm hoping to use "no deployment" to mean paused, and it would be nice to still allow queries in that case

Reply:
When we have subgraph status, we can change `is_deployed` to `is_queriable` and have that mean whatever we want it to mean.